lib/test/TestContract.sol

```diff
@@ -23,4 +28,19 @@ contract TestContract {
         state = _state;
         emit StateSet(_state, _amount);
     }
+
+    function setStateAndPayTokenWithConsumer(uint256 _state, address _erc20, uint256 _amount) public {
+        bool success = tokenConsumer.consume(_erc20, msg.sender, address(this), _amount);
+        if (success) {
+            state = _state;
+            emit StateSet(_state, _amount);
+        }
+    }
+}
+
+contract TokenConsumer {
```

Comment:
I have gone through to separate all contracts into their own files, can we do the same here? There already are a lot of Test* contracts under `lib/test/` so this can also be added there.

Reply:
Done.